vpr/src/route/rr_graph_util.cpp

```diff
@@ -2,14 +2,15 @@
 #include <random>
 #include <algorithm>
 
+#include "rr_graph_util.h"
+
 #include "vtr_memory.h"
 #include "vtr_time.h"
 
 #include "vpr_types.h"
 #include "vpr_error.h"
 
 #include "globals.h"
-#include "rr_graph_util.h"
```

Comment:
Why did you move this?  If there is an include order problem that this solved, change the header order does not solving the underlying problem.

Reply:
Ah, leftover, while implementing this I bumped in some external library linking, but this was not the problem, I'll revert this change

Reply:
I'm aluding to the fact that you may have discovered a problem in the header files.  What exactly did you run into?

Reply:
The issue was actually unrelated to the header files. It was a missing semicolon (typical :) )while implementing the rr_graph_util class, part of the previous code. This missing semicolon was producing this error:

```
In file included from /data/tmp/vtr-verilog-to-routing/vpr/src/route/rr_graph2.h:8:0,
                 from /data/tmp/vtr-verilog-to-routing/vpr/src/route/rr_graph_storage.h:8,
                 from /data/tmp/vtr-verilog-to-routing/vpr/src/base/vpr_context.h:12,
                 from /data/tmp/vtr-verilog-to-routing/vpr/src/base/globals.h:8,
                 from /data/tmp/vtr-verilog-to-routing/vpr/src/base/CheckSetup.cpp:5:
/data/tmp/vtr-verilog-to-routing/vpr/src/route/rr_types.h:6:57: error: invalid declarator before ‘t_pin_to_track_lookup’
 typedef std::vector<vtr::NdMatrix<std::vector<int>, 4>> t_pin_to_track_lookup;
```

Reply:
Very good.